src/library/driftory.ts

```diff
@@ -228,24 +231,28 @@ export default class Driftory {
   }
 
   goToFrame(index: number) {
-    var frame = this.frames[index];
-    var bufferFactor = 0.2;
-    var box = frame.clone();
-
-    box.width *= 1 + bufferFactor;
-    box.height *= 1 + bufferFactor;
-    box.x -= frame.width * bufferFactor * 0.5;
-    box.y -= frame.height * bufferFactor * 0.5;
-
-    this.viewer?.viewport.fitBounds(box);
+    if (this.getFrameIndex() !== index) {
```

Comment:
I wrapped the code in this if statement to prevent it from running the code if the function is called twice for the same `index` value.